src/generic_core/core.ts

```diff
@@ -32,6 +32,7 @@ var bgAppPageChannel = new freedom();
 // Keep track of the current remote instance who is acting as a proxy server
 // for us.
 var remoteProxyInstance : Core.RemoteInstance = null;
+var remoteProxyConnection : Core.RemoteConnection = null;
```

Comment:
The `remoteConnection` variable here is working a bit differently than the `remoteInstance` one.  `remoteInstance` is being used to just store the current instance that we are sending our traffic through while `remoteConnection` is being used to store a single global connection which we can either get or share through (you could actually have both, though the UI experience would be kind of terrible in that case).  I really did not like making this global state, however, I could not come up with a better way to handle both the getting and receiving commands here.

Regarding the commands, the existing ones all took an `instanceId` and had a lot of overhead for setting up the instances.  The other solution I considered was having a check at the top of each command for whether `instanceId` was null and, if so, sending the command to `remoteConnection`.  In the end, I thought that having separate commands without much of a body would look far better than having special cases for each of the existing commands, however, I would be open to more discussion on that.

It also felt a bit more clear to me to have RemoteInstance be build up on top of RemoteConnection rather than having it extend the class.  I would be open to revisiting that.

As far as a design doc, all I have is the UI mockups from Izzie (https://folio.googleplex.com/izzie/uProxy/020615%20-%20Copy+paste#%2FuProxy_copypaste_sketch-02.png%3Fz=half)*